drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
```

Comment:
The test is `//go:build !windows`, should this file be as well?

Reply:
This is used by exec/driver.go which does not have !windows in it, so I think it gets tricky.

I could wrap the use in a function in exec/driver.go like I do with raw_exec and then move that function out to exec/driver_unix.go? And then this could safely be a 100% non-windows file.

Hmm... I'm also debating moving the os-level user getting bits out of this file wholesale, which might change the dynamics of this a bit. In which case, I can leave this as a general validator, then just make the test a _unix test.

Reply:
> This is used by exec/driver.go which does not have !windows in it, so I think it gets tricky.

Right, but I think that's probably ok here? For example, if I later decide to add UID/GID restrictions to the `java` driver, which does run on Windows, I would want the compile to fail if I don't properly split out the behavior so that the Windows build never calls this code.

Reply:
What I could do is remove the uid/gid bits from this and take those as args, and then this file could run on Windows just fine and work - you just wouldn't want to because it probably wouldn't be useful, but nothing would break. And then I could use the windows/unix builds of raw_exec to gate usage (like I already do).

If I am using the validators package to keep the struct for IdRange then I can't make it all unix only because then the import into the Config won't work. I could always make a structs package to get around that though.